packages/ketcher-react/src/script/ui/data/convert/structConverter.ts

```diff
@@ -32,10 +34,29 @@ export function couldBeSaved(
   const rxnArrowsSize = struct.rxnArrows.size;
   const hasRxnArrow = struct.hasRxnArrow();
 
-  if (format === 'smiles' || format === 'smarts') {
-    warnings.push(
-      `Structure contains query properties of atoms and bonds that are not supported in the ${format?.toUpperCase()}. Query properties will not be reflected in the file saved.`,
+  if (format === 'smarts') {
+    const arrayOfAtoms: Array<Atom> = Array.from(struct.atoms.values());
+    const arrayOfBonds: Array<Bond> = Array.from(struct.bonds.values());
+    console.log(arrayOfAtoms);
+
+    const hasAtomUnsupportedProperties = arrayOfAtoms.some(
+      (atom) =>
+        atom.radical ||
+        atom.unsaturatedAtom ||
+        atom.exactChangeFlag ||
+        atom.invRet,
+    );
+    const hasBondUnsupportedProperties = arrayOfBonds.some(
```

Comment:
Maybe change these to something like "bondHasUnsupportedProperties", "atomHasUnsupportedProperties"? Would be more grammatically correct, but I'm not sure what the convention with has/is is on the project.

Reply:
or "bondsHaveUnsupportedProperties" since it's an array.

Reply:
Done